src/types.rs

```diff
@@ -246,14 +243,33 @@ pub struct RecordedEvent {
     /// Representing the user-defined metadata associated with this event.
     pub custom_metadata: Bytes,
 
-    /// Indicates wheter the content is internally marked as JSON.
-    pub is_json: bool,
-
     /// An event position in the $all stream.
     pub position: Position,
 }
 
 impl RecordedEvent {
+    /// Type of this event.
+    pub fn event_type(&self) -> &str {
```

Comment:
I rather store that information I the struct itself because even if the cost is minimal, it is still a cost.

Reply:
Different tradeoffs, I suppose.  My rationale is that the caller can call it once lazily and cache it themselves.  Pre-caching forces the engine to always pay the upfront cost of branching and also memory on the struct.  For our uses cases, we rely on the filtering of the streams in eventstore so it is very often we already know what the event type is on stream listeners.  However, I understand this is not everyone's use case and perhaps it is simpler to just leave it cached and processed by default

Reply:
@YoEight reverted